lifetimes/models/beta_geo_model.py

```diff
@@ -0,0 +1,350 @@
+            phi_prior = pm.Uniform('phi', lower=0.0, upper=1.0)
+            kappa_prior = pm.Pareto('kappa',alpha=1,m=1.5)
+
+            # Beta parameters.
+            a = pm.Deterministic("a", phi_prior*kappa_prior)
+            b = pm.Deterministic("b", (1.0-phi_prior)*kappa_prior)
+
+            logp = pm.Potential('loglike', self._loglike(self.frequency, self.recency, self.T, a, b, alpha_prior, r_prior))
+        
+        return self.model
+
+    def _loglike(
+        self, 
+        frequency: npt.ArrayLike, 
+        recency: npt.ArrayLike, 
+        T: at.var.TensorVariable, 
+        a: at.var.TensorVariable, 
+        b: at.var.TensorVariable, 
+        alpha: at.var.TensorVariable, 
+        r: at.var.TensorVariable,
+        testing: bool =False
+        ) -> Union[Tuple[at.var.TensorVariable],at.var.TensorVariable]:
+        """Log-likelihood function to estimate model parameters for entire population of customers.
+
+        This function was originally introduced in equation 6 of [2]_, and reformulated in section 7 of [3]_
+        to avoid numerical errors for customers who have made large numbers of transactions. 
+        More information can be found in [4]_.
+
+        Parameters
+        ----------
+        frequency: int
+            Total number of transactions for each customer.
+        recency: float64
+            Path where to save model.
+        T: float64
+            Total date periods for each customer.
+        a: aesara TensorVariable
+            Tensor for 'a' parameter of Beta distribution.
+        b: aesara TensorVariable
+            Tensor for 'b' parameter of Beta distribution.
+        alpha: aesara TensorVariable
+            Tensor for 'beta' parameter of Gamma distribution.
+        r: aesara TensorVariable
+            Tensor for 'alpha' parameter of Gamma distribution. 
+        testing: bool
+            Testing flag for term validation. Do not use in production.
+
+        Returns
+        ----------
+        loglike: aesara TensorVariable
+            Log-likelihood value for self._model().
+
+        References
+        ----------
+        .. [2] Fader, Peter S., Bruce G.S. Hardie, and Ka Lok Lee (2005a),
+        "Counting Your Customers the Easy Way: An Alternative to the
+        Pareto/NBD Model," Marketing Science, 24 (2), 275-84.
+        .. [3] http://brucehardie.com/notes/027/bgnbd_num_error.pdf
+        .. [4] http://brucehardie.com/notes/004/
+        """
+
+        # Recast inputs as Aesara tensor variables
+        x = at.as_tensor_variable(frequency)
+        t_x = at.as_tensor_variable(recency)
+        T = at.as_tensor_variable(T)
+
+        x_zero = at.where(x>0, 1, 0)
+
+        # Refactored for numerical error
+        d1 = at.gammaln(r + x) - at.gammaln(r) + at.gammaln(a + b) + at.gammaln(b + x) - at.gammaln(b) - at.gammaln(a + b + x)
+        d2 = r * at.log(alpha) - (r + x) * at.log(alpha + t_x)
+        c3 = ((alpha + t_x)/(alpha + T))**(r+x)
+        c4 = a/(b+x-1)
```

Comment:
We should have a linter for code style right? Black? So that the spaces are consistent. `c4 = a / (b + x -1)`

Reply:
There are many places where we would need to fix a bit the style (e.g spaces). We can use black and flake8. These linters should be added in the CI/CD pipeline. I could support setting this up :)

Reply:
That would be great! I've been dragging my feet on setting up CI/CD due to evaluating the myriad of options available, but it's one of my goals.